subsys/modbus/modbus_serial.c

```diff
@@ -43,6 +43,14 @@ static void modbus_serial_tx_off(struct modbus_context *ctx)
 {
 	struct modbus_serial_config *cfg = ctx->cfg;
 
+	/* Must wait till the transmission is complete or
+	 * tranceiver could be disabled before all data has
+	 * been transmitted and message will be corrupted.
+	 */
+	while (!uart_irq_tx_complete(cfg->dev)) {
```

Comment:
We should not block here as it is executed in ISR context. Additionally it looks like the drivers behave differently, it does not work as expected on FRDM-K64F. You use it with a RS-485 bus transceiver, right?

Reply:
@jfischer-no That is correct we use a RS-485 driver. 

It is possible this can be fixed in the uart driver.

I will spend some time testing that the sam driver behaves according to what is described here: https://docs.zephyrproject.org/latest/doxygen/html/group__uart__interrupt.html

Reply:
okay, I have considered to release DE GPIO delayed. I will try to find a solution in the next days. Overall better RS-485 abstraction would be desirable.